### PR TITLE
fixes for compilation under Visual Studio

### PR DESCRIPTION
**Pull request by nlhepler**
GCC pragmas aren't available (and visibility=hidden by default w/ VS), and __declspec *must* be all the way to the left.

### CODE REVIEW

File: tsne/bh_sne_src/sptree.h [hunk 1]
```diff
@@ -136,5 +138,8 @@ struct SPTree<0> {
   enum { no_children = 1 };
 };
 
+#ifdef __GNUC__
 #pragma GCC visibility pop
 #endif
+
+#endif
```
Comment (adam-azarchs) [inline tsne/bh_sne_src/sptree.h]: This probably should have always been there but before there were other `#if` in here there wasn't anything to be confused about.
```suggestion
#endif  // !defined(SPTREE_H)
```

File: tsne/bh_sne_src/tsne.h [hunk 2]
```diff
@@ -53,7 +53,7 @@
```
Comment (adam-azarchs) [inline tsne/bh_sne_src/tsne.h:61]: I think in windows we also need to define it as `dllimport` when included from places which are not this binary.  If we're only building this into a binary, not as a dll on windows, then we shouldn't be using `dllexport`.

Comment (nlhepler) [reply tsne/bh_sne_src/tsne.h:61]: Given how we're using this, it wasn't strictly necessary. But I did it anyway.

**Comment by nlhepler** [comment]
I'm going to go ahead and merge this, if there are additional changes you would like I can address them in a follow-up.

**Pull request merged**